# Задание 0

<p>Сколько существует 5-разрядных восьмеричных чисел, в&nbsp;которых каждая цифра, кроме первой и&nbsp;последней, окружена цифрами с&nbsp;различной чётностью?
Цифры в&nbsp;числе не&nbsp;повторяются.</p>

In [ ]:
from itertools import permutations, product

s = "01234567"

ans = 0
for i in product(s, repeat=5):
    if i[0] == "0":
        continue
    if len(set(i)) != 5:
        continue
    if (
        int(i[0]) % 2 != int(i[2]) % 2
        and int(i[1]) % 2 != int(i[3]) % 2  
        and int(i[2]) % 2 != int(i[4]) % 2
    ):
        ans += 1
        print(i)
    # print(val, i)

ans

('1', '0', '2', '3', '5')
('1', '0', '2', '3', '7')
('1', '0', '2', '5', '3')
('1', '0', '2', '5', '7')
('1', '0', '2', '7', '3')
('1', '0', '2', '7', '5')
('1', '0', '4', '3', '5')
('1', '0', '4', '3', '7')
('1', '0', '4', '5', '3')
('1', '0', '4', '5', '7')
('1', '0', '4', '7', '3')
('1', '0', '4', '7', '5')
('1', '0', '6', '3', '5')
('1', '0', '6', '3', '7')
('1', '0', '6', '5', '3')
('1', '0', '6', '5', '7')
('1', '0', '6', '7', '3')
('1', '0', '6', '7', '5')
('1', '2', '0', '3', '5')
('1', '2', '0', '3', '7')
('1', '2', '0', '5', '3')
('1', '2', '0', '5', '7')
('1', '2', '0', '7', '3')
('1', '2', '0', '7', '5')
('1', '2', '4', '3', '5')
('1', '2', '4', '3', '7')
('1', '2', '4', '5', '3')
('1', '2', '4', '5', '7')
('1', '2', '4', '7', '3')
('1', '2', '4', '7', '5')
('1', '2', '6', '3', '5')
('1', '2', '6', '3', '7')
('1', '2', '6', '5', '3')
('1', '2', '6', '5', '7')
('1', '2', '6', '7', '3')
('1', '2', '6', '7', '5')
('1', '3', '0', '2', '5')
('1', '3', '0', '2', '7')
('1', '3', '

1008

<h3><details>
	<summary>Ответ</summary>
    1008
</h3>

# Задание 1

<p>Определите количество шестизначных семеричных чисел, в&nbsp;записи которых ровно одна цифра&nbsp;0, а&nbsp;количество остальных чётных цифр чётно.</p>

In [12]:
from itertools import permutations, product

s = '0123456'

# for i in permutations(s, r=5):
#     print(i)


ans = 0
for i in product(s, repeat=6):
    if i[0] == '0':
        continue

    if i.count("0") != 1:
        continue
    
    cnt = 0
    
    for j in '0246':
        cnt += i.count(j)
        
    if cnt % 2 != 0:
        continue
    
    ans += 1

ans
    

19440

<h3><details>
	<summary>Ответ</summary>
    19440
</h3>

# Задание 2

<div class="Markdown-module__Markdown--f5ALc TasksBankTaskCard-module__TaskCardContentMarkdown--hsXqP"><p>Все&nbsp;пятибуквенные слова, в&nbsp;составе которых могут быть только русские буквы Д, О, К, Л, А, записаны в&nbsp;алфавитном порядке и&nbsp;пронумерованы, начиная с&nbsp;1.
Ниже приведено начало списка.</p>
<ol>
<li>ААААА</li>
<li>ААААД</li>
<li>ААААК</li>
<li>ААААЛ</li>
<li>ААААО</li>
<li>АААДА<br>
...</li>
</ol>
<p>Сколько слов будет расположено <strong>между</strong> словами ЛОДКА и&nbsp;ОКЛАД (включая эти слова), которые соответствуют требованиям: в&nbsp;середине слова стоит буква&nbsp;Д, слово содержит ровно две буквы&nbsp;О, вторая с&nbsp;конца буква&nbsp;— согласная.</p></div>

In [17]:
from itertools import permutations, product

s = 'ADKLO'

# for i in permutations(s, r=5):
#     print(i)


ans = 0

f = 0

for i in product(s, repeat=5):
    
    if ''.join(i) == "LODKA":
        f = 1
    
    cond1 = 0
    if i[2] == "D":
        cond1 = 1
    
    cond2 = 0
    if i.count("O") == 2:
        cond2 = 1
    
    cond3 = 0
    
    if i[-2] in "DKL":
        cond3 = 1
        
    if cond1 and cond2 and cond3 and f:
        ans += 1
    
    if ''.join(i) == "OKLAD":
        f = 0
    

ans
    

11

<h3><details>
	<summary>Ответ</summary>
    11
</h3>

# Задание 3

<div class="Markdown-module__Markdown--f5ALc TasksBankTaskCard-module__TaskCardContentMarkdown--hsXqP"><p>Все&nbsp;пятибуквенные слова, в&nbsp;составе которых могут быть только русские буквы С, В, О, Б, Д, А, записаны в&nbsp;алфавитном порядке и&nbsp;пронумерованы, начиная с&nbsp;1.</p>
<p>Ниже приведено начало списка.</p>
<ol>
<li>ААААА</li>
<li>ААААБ</li>
<li>ААААВ</li>
<li>ААААД</li>
<li>ААААО</li>
<li>ААААС</li>
<li>АААБА<br>
...</li>
</ol>
<p>Сколько слов, в&nbsp;которых все буквы различны, будет расположено <strong>между</strong> словами ОСОБА и&nbsp;СДОБА (исключая эти слова)?</p></div>

<h3><details>
	<summary>Ответ</summary>
    92
</h3>

# Задание 4

<p>Сколько существует десятичных пятизначных чисел, <strong>не&nbsp;содержащих</strong> в&nbsp;своей записи цифру&nbsp;5, в&nbsp;которых все цифры различны и&nbsp;никакие две чётные или две нечётные цифры не&nbsp;стоят рядом?</p>

<h3><details>
	<summary>Ответ</summary>
    1056
</h3>

# Задание 5

<p>Определите количество шестизначных чисел, записанных в&nbsp;шестеричной системе счисления, в&nbsp;записи которых ровно одна цифра 2, при этом никакая нечётная цифра не&nbsp;стоит рядом с&nbsp;цифрой&nbsp;2.</p>

<h3><details>
	<summary>Ответ</summary>
    3700
</h3>

# Задание 6

<div class="Markdown-module__Markdown--f5ALc TasksBankTaskCard-module__TaskCardContentMarkdown--hsXqP"><p>Все&nbsp;пятибуквенные слова, в&nbsp;составе которых могут быть только русские буквы Л, А, Й, М, записаны в&nbsp;алфавитном порядке и&nbsp;пронумерованы начиная с&nbsp;1.</p>
<p>Ниже приведено начало списка.</p>
<ol>
<li>
<p>ААААА</p>
</li>
<li>
<p>ААААЙ</p>
</li>
<li>
<p>ААААЛ</p>
</li>
<li>
<p>ААААМ</p>
</li>
<li>
<p>АААЙА<br>
…</p>
</li>
</ol>
<p>Под&nbsp;каким номером в&nbsp;списке идёт последнее слово, которое содержит не&nbsp;более одной буквы М&nbsp;и&nbsp;не&nbsp;содержит букв&nbsp;Л, стоящих рядом?</p></div>

<h3><details>
	<summary>Ответ</summary>
    922
</h3>

# Задание 7

<div class="Markdown-module__Markdown--f5ALc TasksBankTaskCard-module__TaskCardContentMarkdown--hsXqP"><p>Все&nbsp;пятибуквенные слова, в&nbsp;составе которых могут быть только русские буквы Ц, А, П, Л, Я, записаны в&nbsp;алфавитном порядке и&nbsp;пронумерованы начиная с&nbsp;1.</p>
<p>Ниже приведено начало списка.</p>
<ol>
<li>
<p>ААААА</p>
</li>
<li>
<p>ААААЛ</p>
</li>
<li>
<p>ААААП</p>
</li>
<li>
<p>ААААЦ</p>
</li>
<li>
<p>ААААЯ</p>
</li>
<li>
<p>АААЛА<br>
…</p>
</li>
</ol>
<p>Под&nbsp;каким номером в&nbsp;списке идёт первое слово, которое содержит не&nbsp;более одной буквы А, ровно две буквы Ц, не&nbsp;содержит ни&nbsp;одной буквы&nbsp;Л?</p></div>

<h3><details>
	<summary>Ответ</summary>
    319
</h3>

# Задание 8

<div class="Markdown-module__Markdown--f5ALc TasksBankTaskCard-module__TaskCardContentMarkdown--hsXqP"><p>Вася составляет шестибуквенные слова, в&nbsp;которых могут быть использованы только буквы В,&nbsp;И,&nbsp;Ш,&nbsp;Н,&nbsp;Я, причём буква В&nbsp;используется не&nbsp;более одного раза. Каждая из&nbsp;других допустимых букв может встречаться в&nbsp;слове любое количество раз или не&nbsp;встречаться совсем. Слово не&nbsp;должно начинаться с&nbsp;буквы Ш&nbsp;и&nbsp;оканчиваться гласными буквами. Словом считается любая допустимая последовательность букв, не&nbsp;обязательно осмысленная.</p>
<p>Сколько существует таких слов, которые может написать Вася?</p></div>

<h3><details>
	<summary>Ответ</summary>
    4352
</h3>

# Задание 9

<p>Сколько существует шестеричных семизначных чисел, содержащих в&nbsp;своей записи ровно одну цифру 2, при этом никакая чётная цифра не&nbsp;стоит рядом с&nbsp;цифрой&nbsp;2?</p>

<h3><details>
	<summary>Ответ</summary>
    40500
</h3>

# Задание 10

<p>Сколько существует тринадцатеричных шестизначных чисел, не содержащих в своей записи более одной цифры 5, в которых никакие две нечётные цифры не стоят рядом?</p>

<h3><details>
	<summary>Ответ</summary>
    1666784
</h3>

# Задание 11

Сколько существует девятиразрядных десятичных чисел таких, в которых есть хотя бы 3 различные цифры?

In [ ]:
111_111_111, ..., 999_999_999 = 9

In [ ]:
cnt = 0
# 1, ('0', '0', '0', '0', '0', '0', '0', '0') 
for i in product('01', repeat=8):
    
    if len(set(i)) != 1:
        cnt += 1
        # print(i)
# 255
print(cnt)
# 10, 20, 30, 40
# 255 * 9 = 2295

#_________________________________________________

# 12, 13, ... 8
# 23, 24 ... 7
# 34, 35 .. 6
# ...

print(900_000_000 - ((8 * 9 // 2) * 510 + 255 * 9 + 9))

# (8 * 9 // 2) * 510 -- количество чисел из пар цифр 1, 2, 3, 4, 5 ...
# 255 * 9 -- количество чисел из пар цифр 1 0, 2 0, 3 0, 4 0, 5 0 ...
# 9 -- количество чисел состоящих из 1 цифры
cnt = 0
for i in product('01', repeat=9):
    
    if len(set(i)) != 1:
        cnt += 1
        # print(i)
# 510

print(cnt)




254
899979336
510


In [ ]:
101_101_101, 010_111_000

In [8]:
cnt = 0

for i in range(100_000_000, 999_999_999):
    if len(set(str(i))) >= 3:
        cnt += 1

<image src="./src/2024-11-25_17-29-04.png">

In [9]:
cnt

899979336

In [11]:
cnt = 0

for i in range(100_000_000, 999_999_999):
    d = {}
    for j in str(i):
        if j not in d:
            d[j] = 1
    # print(d, len(d))
    if len(d) >= 3:
        cnt += 1

<image src="./src/2024-11-25_17-41-12.png">

In [12]:
cnt

899979336

# Задание 12

<div class="Markdown-module__Markdown--f5ALc TasksBankTaskCard-module__TaskCardContentMarkdown--hsXqP"><p>Григорий составляет буквенные последовательности путем перестановки букв из&nbsp;слова НОСОЧЕЧКИ.</p>
<p>Сколько различных последовательностей может составить Григорий, если известно, что гласные и&nbsp;согласные буквы чередуются?</p></div>

In [ ]:
for p in permutations("NOSOCECKI"):
    print(p)

<h3><details>
	<summary>Ответ</summary>
    720
</h3>

# Задание 13

<div class="Markdown-module__Markdown--f5ALc TasksBankTaskCard-module__TaskCardContentMarkdown--hsXqP"><p>Все шестибуквенные слова, в&nbsp;составе которых могут быть только русские буквы Б,&nbsp;Ю,&nbsp;Д,&nbsp;Ж,&nbsp;Е,&nbsp;Т,&nbsp;Н,&nbsp;И,&nbsp;К, записаны в&nbsp;алфавитном порядке и&nbsp;пронумерованы, начиная с&nbsp;1.</p>
<p>Ниже приведено начало списка.</p>
<ol>
<li>ББББББ</li>
<li>БББББД</li>
<li>БББББЕ</li>
<li>БББББЖ</li>
<li>БББББИ</li>
<li>БББББК</li>
<li>БББББН</li>
<li>БББББТ</li>
<li>БББББЮ<br>
…</li>
</ol>
<p>Какова сумма номеров слов, расположенных до&nbsp;слова БЮДЖЕТ (не&nbsp;включая это слово), в&nbsp;которых никакие две гласные буквы не&nbsp;стоят рядом
и&nbsp;никакие две согласные буквы не&nbsp;стоят рядом?</p></div>

<h3><details>
	<summary>Ответ</summary>
    18575058
</h3>

# Задание 14

<p>Сколько 7 разрядных двенадцатеричных чисел существует, в&nbsp;которых разряды, делящиеся без остатка на&nbsp;3, и&nbsp;разряды, не&nbsp;делящиеся без остатка на&nbsp;3, чередуются?</p>

In [27]:
12 ** 7 / 5_000_000

7.1663616

<h3><details>
	<summary>Ответ</summary>
    360448
</h3>